runeq/resources/project.py

```diff
@@ -0,0 +1,182 @@
+"""
+Fetch metadata about projects.
+
+"""
+
+from typing import Iterable, Optional, Type
+
+from .client import GraphClient, global_graph_client
+from .common import ItemBase, ItemSet
+
+
+class Project(ItemBase):
+    """
+    Metadata for an project.
+
+    """
+
+    def __init__(
+        self,
+        id: str,
+        title: str,
+        status: str,
+        type: str,
+        started_at: float,
+        updated_at: float,
+        created_at: float,
+        created_by: str,
+        updated_by: str,
+        **attributes
+    ):
+        """
+        Initialize with data.
+
+        Args:
+            id: ID of the project
+            title: Human-readable name
+            created_at: When the project was created (unix timestamp)
+            status: Status of the project,
+            type: Type of project,
+            started_at: Time the project started (unix timestamp),
+            created_at: Time the project was created (unix timestamp),
+            created_by: UserID of who created the project,
+            updated_at: Time the project was last updated (unix timestamp),
+            updated_by: UserID of who updated the project,
+            **attributes: Other attributes associated with the project
+
+        """
+        # Update to include required attributes
+        self.title = title
+        self.created_at = created_at
+        self.started_at = started_at
+        self.updated_at = updated_at
+        self.created_by = created_by
+        self.updated_by = updated_by
+        self.status = status
+        self.type = type
+
+        super().__init__(
+            id=id,
+            title=title,
+            created_at=created_at,
+            started_at=started_at,
+            updated_at=updated_at,
+            created_by=created_by,
+            updated_by=updated_by,
+            status=status,
+            type=type,
+            **attributes,
+        )
+
+
+class ProjectSet(ItemSet):
+    """
+    A collection of Projects.
+
+    """
+
+    def __init__(self, items: Iterable[Project] = ()):
+        """
+        Initialize with Projects.
+
+        """
+        super().__init__(items=items)
+
+    @property
+    def _item_class(self) -> Type[ItemBase]:
+        """
+        Instance type of items in this set.
+
+        """
+        return Project
+
+
+def get_project(
+    project_id: str, client: Optional[GraphClient] = None
+) -> Project:
+    """
+    Get the project with the specified ID.
+
+    Args:
+        project_id: Project ID
+        client: If specified, this client is used to fetch metadata from the
+            API. Otherwise, the global GraphClient is used.
+
+    """
+    client = client or global_graph_client()
+    query = """
+        query getProject($project_id: ID) {
+            project (projectId: $project_id) {
```

Comment:
Hmm does this query work? I ran it locally and got `"Unknown argument 'projectId' on field 'Query.project'."`. It looks like this change needs to be made:
```suggestion
            project (id: $project_id) {
```

Reply:
I found it helpful for me to run the queries with my credentials in `initialize()` after updating the queries to catch errors like this.

Reply:
👍🏻